sdk/file_sdk/XMPFileFormat/xmp_file_format.cpp

```diff
@@ -38,10 +43,30 @@ void XMPFileFormat::Commit(shared_ptr<IStream> outputStream, string& newExtensio
   while (readCount > 0);
 
   outputStream->Seek(0);
-  XMPHelper::GetInstance().SetTags(outputStream, mTags);
+  XMPHelper::GetInstance().SetProperties(outputStream, mPropertiesToAdd, mkeysToRemove);
 
   newExtension = mExtension;
 }
 
+void XMPFileFormat::UpdateProperties(const std::vector<std::pair<std::string, std::string>>& propertiesToAdd, const std::vector<std::string>& keysToRemove)
+{
+  mPropertiesToAdd = propertiesToAdd;
+  mkeysToRemove = keysToRemove;
+
+  if(mHasProperties) {
```

Comment:
Is this only for the point of efficiency when calling GetProperties after UpdateProperties?

Reply:
the Boolean? yes, it is

---
In reply to: [140659334](https://github.com/AzureAD/rms-sdk-for-cpp/pull/168#discussion_r140659334) [](ancestors = 140659334)

Reply:
Do you really want to do all this, even if GetProperties will never be called again?